hazelcast/src/main/java/com/hazelcast/map/impl/operation/QueryOperation.java

```diff
@@ -115,12 +113,41 @@ public void run() throws Exception {
         checkPartitionStateChanges(partitionService, partitionStateVersion);
     }
 
+    private void fullTableScan(Collection<Integer> initialPartitions, GroupProperties groupProperties)
+            throws InterruptedException, ExecutionException {
+        if (pagingPredicate != null) {
+            runParallelForPaging(initialPartitions);
+        } else {
+            boolean parallelEvaluation = groupProperties.QUERY_PREDICATE_PARALLEL_EVALUATION.getBoolean();
+            if (parallelEvaluation) {
+                runParallel(initialPartitions);
+            } else{
+                runSingleThreaded(initialPartitions);
+            }
+        }
+    }
+
     private void checkPartitionStateChanges(InternalPartitionService partitionService, int partitionStateVersion) {
         if (partitionStateVersion != partitionService.getPartitionStateVersion()) {
             getLogger().info("Partition assignments changed while executing query: " + predicate);
         }
     }
 
+    protected void runSingleThreaded(final Collection<Integer> initialPartitions) throws InterruptedException, ExecutionException {
+        final NodeEngine nodeEngine = getNodeEngine();
+        final ExecutorService executor
+                = nodeEngine.getExecutionService().getExecutor(ExecutionService.QUERY_EXECUTOR);
+
+        for (Integer partitionId : initialPartitions) {
+            MapContextQuerySupport mapContextQuerySupport = mapService.getMapServiceContext()
+                    .getMapContextQuerySupport();
+            Collection<QueryableEntry> queryableEntries = mapContextQuerySupport.queryOnPartition(name, predicate, partitionId);
```

Comment:
Note to myself: We may have to modify this like `runParallel()` if the unbounded return values feature get's merged.

Will be something like this (not tested):
```
        Collection<Collection<QueryableEntry>> returnedResults = mapContextQuerySupport.queryOnPartition(name, predicate, partitionId);
        for (Collection<QueryableEntry> returnedResult : returnedResults) {
            if (returnedResult == null) {
                continue;
            }
            if (!addQueryEntriesToResult(returnedResult)) {
                break;
            }
        }
```